src/client/datascience/raw-kernel/liveshare/hostRawNotebookProvider.ts

```diff
@@ -169,10 +170,15 @@ export class HostRawNotebookProvider
                     // this code as well as the code where we initialize the interpreter via a hack.
                     // This is used to check if there are situations under which this is possible & to safeguard against it.
                     // The only real world scenario is when users do not install Python (which we cannot prevent).
-                    (kernelConnection as any).interpreter = this.kernelService.findMatchingInterpreter(
+                    const readWriteConnection = kernelConnection as ReadWrite<KernelConnectionMetadata>;
+                    readWriteConnection.interpreter = await this.kernelService.findMatchingInterpreter(
                         kernelConnection.kernelSpec,
                         cancelToken
                     );
+                    if (readWriteConnection.kind === 'startUsingKernelSpec') {
+                        readWriteConnection.kernelSpec.interpreterPath =
+                            readWriteConnection.kernelSpec.interpreterPath || readWriteConnection.interpreter?.path;
```

Comment:
Found a bug, fixing.

Reply:
Doesn't this break the idea of the connection metadata being immutable? Should this happen during creation of the connection metadata?

Reply:
Yes that's correct, added telemetry for this to check if this case it hit.
Sorry, the message was generic, i didn't find a bug, i found a similar bug in another place & then found this code.
The telemetry will help us determine if this case is ever hit, & it shouldn't

Reply:
The other place was kernel selections where this is built & we initialize this. & when i searched i found this other place & copied the same code here as well.